src/plugins/intel_cpu/src/transformations/cpu_opset/common/pass/move_readvalue_inputs_to_subgraph.cpp

```diff
@@ -0,0 +1,194 @@
+
+        if (inputs.size() == 0 || corresponding_assign == nullptr || subgraph.size() == 0) {
+            return false;
+        }
+
+        // ====== Debug log: ==================================
+        {
+            std::cout << "============================================1" << std::endl;
+            std::cout << "  == Found readvalue = " << readvalue->get_friendly_name() << std::endl;
+            std::cout << "  == Found inputs = ";
+            for (auto inp : inputs) {
+                std::cout << inp->get_friendly_name() << ", ";
+            }
+            std::cout << std::endl;
+            std::cout << "  == Found subgraph = ";
+            for (auto nd : subgraph) {
+                std::cout << nd->get_friendly_name() << ", ";
+            }
+            std::cout << std::endl;
+            std::cout << "  == Found corresponding_assign = " << corresponding_assign->get_friendly_name() << std::endl;
+            std::cout << "============================================2" << std::endl;
+        }
+
+        auto new_rv = std::make_shared<ov::intel_cpu::ReadValueWithSubgraphNode>(readvalue->get_variable());
+
+        auto is_in_subgraph = [&subgraph](std::shared_ptr<ov::Node> n) {
+            for (auto& sub : subgraph) {
+                if (n->get_friendly_name() == sub->get_friendly_name()) {
+                    return true;
+                }
+            }
+            return false;
+        };
+        // Subgraph's input
+        auto params = ParameterVector{};
+        for (auto inp : inputs) {
+            auto param = std::make_shared<ov::op::v0::Parameter>(inp->get_element_type(), inp->get_output_partial_shape(0));
+            params.push_back(param);
+            for (const auto& child : inp->get_output_target_inputs(0)) {
+                if (is_in_subgraph(child.get_node()->shared_from_this())) {
+                    std::cout << "-----> child.replace_source_output, child="
+                              << child.get_node()->shared_from_this()->get_friendly_name() << std::endl;
+                    child.replace_source_output(param);
+                }
+            }
+        }
+
+        // Subgraph's output
+        auto last_node = readvalue->get_input_node_shared_ptr(0);
+        // auto rv_subgraph = readvalue->clone_with_new_inputs({last_node});
+        // auto assign_subgraph = corresponding_assign->clone_with_new_inputs({rv_subgraph});
+        auto output = std::make_shared<ov::op::v0::Result>(last_node);
+        auto func = std::make_shared<Model>(ov::ResultVector({output}), params);
+#if USE_SUBMODEL
+        std::cout << "------1: " << std::endl;
+        auto submodel = std::make_shared<SubModel>(func);
+        std::cout << "------2: " << std::endl;
+        for (size_t i = 0; i < inputs.size(); i++) {
+            submodel->set_invariant_input(params[i], inputs[i]->output(0));
+        }
+        std::cout << "------3: " << std::endl;
+        // submodel->set_body_outputs({output});
```

Comment:
Hi @EgorDuplensky , I am trying to integrate your new OP(`SubModel`), I have merged your [PR 25385](https://github.com/openvinotoolkit/openvino/pull/25385) into current PR. Here I have a question.
I don't know if I need to call `set_body_outputs({output});` here.

If I call it, `validate_and_infer_types` will be called in the `set_body_outputs`, but `void SubModel::validate_and_infer_types()` also call `set_output_size`, it will trigger set output twice.  https://github.com/openvinotoolkit/openvino/pull/25385/files#diff-12815bb2cbf8df4ae1c5cac20015ede2743f93792d043da4e570f70cb9d0c768R51

If I don't call it, but I seem can't  set output for my node `ov::intel_cpu::ReadValueWithSubgraphNode`.

Could you please help comment it? and what's your expected?

Reply:
Sorry @EgorDuplensky ,Please ignore it.
Just synced with @maxnick , I misunderstand @maxnick 's idea. I will try to use new intel_cpu::Graph interface introduced by you.